crates/rethnet_eth/src/remote/cacheable_method_invocation.rs

```diff
@@ -0,0 +1,154 @@
+use crate::remote::methods::{GetLogsInput, MethodInvocation, Percentiles, TransactionInput};
+use crate::remote::BlockSpec;
+use crate::U256;
+use revm_primitives::{Address, B256};
+
+/// These method invocations are hashable and can be potentially cached.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+// Using repr(u8) to let us assign a number to each variant which is important to make sure hashing
+// remains consistent if new variants are added or if variants are reordered.
+#[repr(u8)]
+pub(super) enum CacheableMethodInvocation<'a> {
+    /// eth_call
+    Call(&'a TransactionInput, &'a Option<BlockSpec>) = 0,
+    /// eth_chainId
+    ChainId() = 1,
+    /// eth_estimateGas
+    EstimateGas(&'a TransactionInput, &'a Option<BlockSpec>) = 2,
+    /// eth_feeHistory
+    FeeHistory(
+        /// block count
+        &'a U256,
+        /// newest block
+        &'a BlockSpec,
+        /// reward percentiles
+        &'a Percentiles,
+    ) = 3,
+    /// eth_getBalance
+    GetBalance(&'a Address, &'a Option<BlockSpec>) = 4,
+    /// eth_getBlockByNumber
+    GetBlockByNumber(
+        &'a BlockSpec,
+        /// include transaction data
+        bool,
+    ) = 5,
+    /// eth_getBlockByHash
+    GetBlockByHash(
+        /// hash
+        &'a B256,
+        /// include transaction data
+        bool,
+    ) = 6,
+    /// eth_getBlockTransactionCountByHash
+    GetBlockTransactionCountByHash(&'a B256) = 7,
+    /// eth_getBlockTransactionCountByNumber
+    GetBlockTransactionCountByNumber(&'a BlockSpec) = 8,
+    /// eth_getCode
+    GetCode(&'a Address, &'a Option<BlockSpec>) = 9,
+    /// eth_getLogs
+    GetLogs(&'a GetLogsInput) = 11,
+    /// eth_getStorageAt
+    GetStorageAt(
+        &'a Address,
+        /// position
+        &'a U256,
+        &'a Option<BlockSpec>,
+    ) = 12,
+    /// eth_getTransactionByBlockHashAndIndex
+    GetTransactionByBlockHashAndIndex(&'a B256, &'a U256) = 13,
+    /// eth_getTransactionByBlockNumberAndIndex
+    GetTransactionByBlockNumberAndIndex(&'a U256, &'a U256) = 14,
+    /// eth_getTransactionByHash
+    GetTransactionByHash(&'a B256) = 15,
+    /// eth_getTransactionCount
+    GetTransactionCount(&'a Address, &'a Option<BlockSpec>) = 16,
+    /// eth_getTransactionReceipt
+    GetTransactionReceipt(&'a B256) = 17,
+    /// net_version
+    NetVersion() = 18,
+}
+
+/// Error type for [`CacheableMethodInvocation::try_from`].
+#[derive(thiserror::Error, Debug)]
+#[error("Method is not cacheable: {0:?}")]
+pub(super) struct MethodNotCacheableError(MethodInvocation);
+
+impl<'a> TryFrom<&'a MethodInvocation> for CacheableMethodInvocation<'a> {
+    type Error = MethodNotCacheableError;
+
+    fn try_from(value: &'a MethodInvocation) -> Result<Self, Self::Error> {
+        use CacheableMethodInvocation as CMI;
```

Comment:
Even though I understand the desire to avoid retyping of the same long type name here, we are actively moving away from non-standard acronyms in our codebase to keep it more readable. Even our generic types and lifetimes are becoming full names.

I say non-standard because things like `Eth` or `IO` are completely acceptable.